simulator/src/main/java/com/hedera/block/simulator/config/types/GenerationMode.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.config.types;
+
+/** The GenerationMode enum defines the modes of generation for the block stream. */
+public enum GenerationMode {
+    /** Reads Blocks from a Folder. */
+    FOLDER,
+    /** Generates Blocks from rules */
+    SELF
```

Comment:
What do you think about changing `SELF` to `AD_HOC` or `ADHOC`?

Reply:
What about changing `FOLDER` to `DIR` to align with linux, etc?

Reply:
Again, also inherited those Enums but I like your suggestions, will change Folder for `DIR` and SELF for `ADHOC`